packages/zowe-explorer/src/trees/dataset/DatasetUtils.ts

```diff
@@ -75,14 +75,8 @@ export class DatasetUtils {
         const split = bracket > -1 ? label.substring(0, bracket).split(".", limit) : label.split(".", limit);
         for (let i = split.length - 1; i > 0; i--) {
             for (const [ext, matches] of DS_EXTENSION_MAP.entries()) {
-                for (const match of matches) {
-                    if (match instanceof RegExp) {
-                        if (match.test(split[i])) {
-                            return ext;
-                        }
-                    } else if (match.includes(split[i])) {
-                        return ext;
-                    }
+                if (matches.some((match) => (match instanceof RegExp ? match.test(split[i]) : match === split[i]))) {
```

Comment:
thanks for rewritting 🙏 
Not sure if this makes it more or less readable, but I kind of prefer it this way (less lines of code) 😅 

And I guess ther diff here is ``` `.incluides` -> `===` ```